backend/app/views.py

```diff
@@ -274,4 +278,34 @@ def corpus(request, corpus_id):
     }
 
     return render(request, 'index.html', context)
-    
+
+
+@api_view(['GET'])
+def all_frequency_analyses(request):
+    freq_analysis_objs = FrequencyAnalysis.objects.all()
+    serializer = FrequencyAnalysisSerializer(freq_analysis_objs, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def add_frequency_analysis(request):
+    attributes = request.data
+    corpus_id = attributes['corpus_id']
+    gender_ids = attributes['gender_ids']
+    gender_set = Gender.objects.filter(id__in=gender_ids)
+    frequency_entry = FrequencyAnalysis.objects.filter(corpus__id=corpus_id)\
+        .annotate(num_genders=Count('genders')).filter(num_genders=len(gender_ids))
+    for id in gender_ids:
+        frequency_entry.filter(genders__id=id)
```

Comment:
I am a little confused by the purpose/significance of this block of code, not sure if calling `filter()` for each gender_id is doing what you may think it is doing because isn't a new QuerySet filled with desired entries returned by `filter()` every time you call it. So when you go and call `frequency_entry.exists()`, it will come back as False in most cases because you are filtering by the same parameter each time.

Reply:
Yes, I agree: the `.filter()` method does not act on the `frequency_entry` `QuerySet`, but it returns a new `QuerySet` each time containing a subset of the model instances that `frequency_entry` would contain. Then line 300 acts on the same `QuerySet` as if lines 298-299 weren't there. You could do something like `frequency_entry = frequency_entry.filter(genders__id=id)`*, but there is likely a more elegant way to do this.

*Although this would surely work for a `ForeignKey` relation, things may look slightly different for a `ManyToMany` relation.

Reply:
Ah, I caught the bug and I fixed it. Thank you! We tried looking for a more elegant way to filter the `ManyToMany` relation but the simplest way is first filtering the instances for ones with the same number of genders and then going through the list of genders and ensuring that each of them is in the analysis instance.

Reply:
I think that doing  `frequency_entry = frequency_entry.filter(genders__id=id)`  still doesn't let you check if each gender is in the analysis instance because you're reassigning `frequency_entry` with the filtered `QuerySet` , so after the first iteration of the loop you won't see the other genders because you would have filtered it to have only one gender. If you want to check if each gender is in the analysis, maybe it should be completed in the for loop?

Reply:
What `frequency_entry = frequency_entry.filter(genders__id=id)` does in the for loop is that at every iteration, it checks  to see if the current `id` is in `genders`. For example, let's say we're looking for the genders list `[1, 2]` and in our database we have `[1]`, `[1, 2]`, `[2]`. In the first iteration, we look for the gender id 1 and since both entries in the database has the gender id 1, both of them will be returned in the filter. Now in the second iteration, we look for the gender id 2 in the updated query set and since only `[1, 2]` has the gender id 2, only the entry with `[1, 2]` will be returned.

Reply:
Thanks for clarifying! I misunderstood what you were trying to do, this makes sense now. I thought that you were attempting to look for the occurrence of single gender ids in each iteration.